appserver/jdbc/jdbc-ra/jdbc-core/src/main/java/com/sun/gjc/util/SQLTraceDelegator.java

```diff
@@ -101,7 +101,7 @@ public void setModuleName(String moduleName) {
      */
     public void registerSQLTraceListener(SQLTraceListener listener) {
         if (sqlTraceListeners == null) {
-            sqlTraceListeners = new HashMap<>();
+            sqlTraceListeners = new ConcurrentHashMap<>();
```

Comment:
Shouldn't be also type of variable changed to interface ConcurrentMap to enforce implementations that support parallelism?